src/SyntaxParser/ImportedClassesParser.php

```diff
@@ -25,6 +28,48 @@ public function parseImportedClasses(ContextualToken $firstToken)
         return $importedClasses;
     }
 
+    public function parseImportedClassesFromTokens(Tokens $tokens)
+    {
+        $importedClasses = new ImportedClasses();
+        $namespaceIndex = $tokens->getNextTokenOfKind(0, [[T_NAMESPACE]]);
+        $namespaceEndIndex = $tokens->getNextTokenOfKind($namespaceIndex, [';']);
+        $start = $tokens->getNextMeaningfulToken($namespaceIndex);
+        $end = $tokens->getPrevMeaningfulToken($namespaceEndIndex);
+        $namespace = $tokens->generatePartialCode($start, $end);
+        $importedClasses->setCurrentNamespace($namespace);
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+        $useIndices = $tokensAnalyzer->getImportUseIndexes();
+        foreach ($useIndices as $useIndex) {
+            $className = null;
+            $currentContent = '';
+            for ($i = $useIndex + 1; $i < $tokens->count(); $i++) {
```

Comment:
for inside foreach loop, too complex, move logic that resolved `$className` to separate private method